nixos/modules/services/web-servers/nginx/vhost-options.nix

```diff
@@ -367,4 +386,69 @@ with lib;
       description = "Declarative location config";
     };
   };
+
+  config = let
+   inherit (global-config.services.nginx) package;
+  in {
+    warnings = optional config.enableSSL
+      ''
+        config.services.nginx.virtualHosts."${name}".enableSSL is deprecated,
+        use config.services.nginx.virtualHosts."${name}".onlySSL instead.
+      '';
+
+    assertions = [
+      (let
+        matchedLocations = filterAttrs (n: v: v.root != null && v.alias != null) config.locations;
+      in{
+        assertion = (builtins.length (lib.attrValues matchedLocations)) > 0;
+        message = "Only one of root or alias can be specified on location ${lib.concatStringsSep ", " (attrValues matchedLocations)}.";
```

Comment:
I think we should also mention which vhost this affects now that we have the information.